In [11]:
# Importing libraries
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.naive_bayes import GaussianNB

In [12]:
# Importing dataset
df = pd.read_csv('Iris.csv', index_col=0)

In [13]:
# Looking for columns and rows
df.shape

(150, 5)

This shows that this dataset has 150 records(rows) and 6 features(columns).

In [14]:
# Making encoding dictionary
encoding_dict = {
    "Species" : {"Iris-setosa":0, "Iris-versicolor":1, "Iris-virginica":2}
}

In [15]:
# Replacing the categorical data to numerical data.
df = df.replace(encoding_dict)

In [16]:
# I/O data
X = df.iloc[:,:-1]
y = df.iloc[:,-1]

In [18]:
# Model initialization.
clf = GaussianNB()

In [34]:
# Tuple
score_tuple = tuple()

In [35]:
# Using for loop for getting train test data and model training and evaluation.
for i in np.arange(50,70,5):
    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.18, random_state=i)
    clf.fit(X_train, y_train)
    score_tuple = score_tuple + (clf.score(X_test, y_test),)# append tuple using concatination

In [36]:
# Printing tup
print(score_tuple)

(0.9629629629629629, 0.9629629629629629, 0.9259259259259259, 0.9259259259259259)
